GenderPayGap.WebUI/Controllers/AddOrganisation/AddOrganisationManualAddressController.cs

```diff
@@ -46,8 +46,22 @@ public IActionResult ManualAddress(AddOrganisationManualViewModel viewModel)
                 }
 
                 viewModel.Validate = null; // Required to prevent the next page immediately trying to validate the (empty) address
-                return RedirectToAction("ManualSicCodes", "AddOrganisationManualSicCodes", viewModel);
-
+                if (viewModel.Editing == true)
+                {
+                    viewModel.Editing = null; // To make the url look a bit nicer (the Review page implies we're editing so there's no need for "Editing" in the url)
+                    return RedirectToAction("ManualReview", "AddOrganisationManualReview", viewModel);
+                }
+                else
+                {
+                    if (viewModel.Sector.Value == AddOrganisationSector.Private)
+                    {
+                        return RedirectToAction("ManualSicCodes", "AddOrganisationManualSicCodes", viewModel);
+                    }
+                    else
+                    {
+                        return RedirectToAction("ManualReview", "AddOrganisationManualReview", viewModel);
+                    }
+                }

```

Comment:
It might just be me being picky, but this structure of if/else clauses seems a bit weird to me. Both the check for `viewModel.Editing` and the default behaviour for non-private sectors are to redirect to the `ManualReview` function, with one property differing between them.

To me, it'd make more sense to have

```
if(viewModel.Editing == true || viewModel.Sector.Value == AddOrganisationSector.Private) {
    // change viewModel.Editing if needed
    return RedirectToAction("ManualReview", "AddOrganisationManualReview", viewModel);
}
return RedirectToAction("ManualSicCodes", "AddOrganisationManualSicCodes", viewModel);
```

It's essentially the same thing, but saves a bit of duplicate code?

Reply:
I was trying to communicate why we go to the different pages.
There's 3 distinct cases, so I thought it would be good to have 3 blocks:
* If we're in the "Editing" flow, we go to each page, then back to the "Review" page
* If we're in the "Normal" flow, we go to each page in turn...
  * For private-sector organisations, we need them to add SIC codes, so "SicCodes" is the next page
  * For public-sector organisations, we don't use SIC codes, so the next page is the Review page

2 of the outcomes are the same, but they might well change in future (since we're still deciding on the design of this journey).
I don't want us to assume that these two outcomes are linked, when they're actually quite separate.

It does create quite verbose code, which is a shame.
I've pulled this block (in each of these controllers) out into a separate method, to try to make each method smaller and clearer.
I've also added some comments, to say what each case is for.